google/oauth2/service_account.py

```diff
@@ -569,6 +561,14 @@ def __init__(
         self._quota_project_id = quota_project_id
         self._use_iam_endpoint = False
 
+        if not universe_domain:
+            self._universe_domain = _DEFAULT_UNIVERSE_DOMAIN
+        else:
+            self._universe_domain = universe_domain
+
+        if universe_domain != _DEFAULT_UNIVERSE_DOMAIN:
+            self._use_iam_endpoint = True
```

Comment:
In the with_iam_endpoint method we probably want to check if universe is not a default before overriding in case of False input.

Reply:
added logic to throw because if universe is not default only the iam endpoint id token flow work